### PR TITLE
Fix For When user selects Empty files using File Upload

### PR DESCRIPTION
**Pull request by Yummy-Yums**
### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by Yummy-Yums** [comment]
#2873

File: reflex/.templates/web/utils/state.js [hunk 1]
```diff
@@ -240,7 +240,13 @@ export const applyEvent = async (event, socket) => {
  */
 export const applyRestEvent = async (event, socket) => {
   let eventSent = false;
-  if (event.handler == "uploadFiles") {
+  if (event.handler === "uploadFiles") {
+
+    if (event.payload.files === undefined || event.payload.files.length === 0){
+      // Submit the event over the websocket to trigger the event handler.
+      return await applyEvent(Event(event.name, event.payload.files), socket)
```
Comment (masenf) [inline reflex/.templates/web/utils/state.js:247]: the payload is supposed to be a object, not an array

Comment (Yummy-Yums) [reply reflex/.templates/web/utils/state.js:247]: you mean `return await applyEvent(Event(event.name, event.payload), socket)`

Comment (masenf) [reply reflex/.templates/web/utils/state.js:247]: the problem is that `event.payload` contains keys that do not map to an arbitrary event handler on the backend.

for example, maybe it works if the backend event handler has a `files` arg; but if that arg is named `fools`, then we get an error on the backend about unknown keyword argument.

File: reflex/.templates/web/utils/state.js [hunk 2]
```diff
@@ -381,7 +387,7 @@ export const uploadFiles = async (
 ) => {
   // return if there's no file to upload
   if (files === undefined || files.length === 0) {
-    return false;
+    files = []
```
Comment (masenf) [inline reflex/.templates/web/utils/state.js:390]: what is this change doing? if we end up hitting this code, won't it ultimately throw that "Did not find CR at end of boundary (40)" error in the upload handler?

**Review by masenf — APPROVED** [review]
the only thing that still makes me nervous about this is that, if the handler doesn't specify a default for the `list[rx.UploadFile]` arg, then there will be a traceback printed on the backend, whereas before, this was just a no op.

i think that's okay though. it will still be a no-op, and if they want to avoid the traceback, they can add a default value...

**Comment by Yummy-Yums** [comment]
> the only thing that still makes me nervous about this is that, if the handler doesn't specify a default for the `list[rx.UploadFile]` arg, then there will be a traceback printed on the backend, whereas before, this was just a no op.
> 
> i think that's okay though. it will still be a no-op, and if they want to avoid the traceback, they can add a default value...

i think we can put it in a doc somewhere ? or leave it like that? what dyu suggest ?

**Pull request merged**